src/basic/openssl-util.c

```diff
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#if HAVE_OPENSSL
+
+#include "openssl-util.h"
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(HMAC_CTX*, HMAC_CTX_free);
+
+int hmac(
+        const EVP_MD *alg,
+        const uint8_t *key,
+        int key_len,
+        const uint8_t *msg,
```

Comment:
`const void*` sounds more appropriate? I mean, we might want to HMAC other stuff than just plain bytes...